napari/components/_tests/test_add_layers.py

```diff
@@ -44,14 +39,24 @@ def test_plugin_returns_nothing():
     MagicMock(return_value=[(img,)]),
 )
 def test_viewer_open():
-    """Test that a plugin to returning nothing adds nothing to the Viewer."""
+    """Test that a plugin to returning an image adds stuff to the viewer."""
     viewer = ViewerModel()
     assert len(viewer.layers) == 0
-    viewer.open('mock_path')
+    viewer.open('mock_path.tif')
     assert len(viewer.layers) == 1
+    # The name should be taken from the path name, stripped of extension
+    assert viewer.layers[0].name == 'mock_path'
 
-    viewer.open('mock_path', stack=True)
+    # stack=True also works... and very long names are truncated
+    viewer.open('mock_path_very_long_name.tif', stack=True)
     assert len(viewer.layers) == 2
+    assert viewer.layers[1].name == 'mock_path_very...'
```

Comment:
I don’t love this at all. It’s not a very long truncation length! I can see many people coming across this limit with their images.

Reply:
I'll remove it.  It was the length that fits in the LayerList without truncation... but probably no reason to make special accommodations for that